builder/comp-builder.nix

```diff
@@ -40,9 +40,10 @@ let self =
 , enableDeadCodeElimination ? component.enableDeadCodeElimination
 
 # Options for Haddock generation
-, doHaddock ? component.doHaddock  # Enable haddock and hoogle generation
+, doHaddock ? component.doHaddock  # Not used any more.  Should we use it?
```

Comment:
Indeed, maybe we should provide something like `collectChecks` which does this. Or maybe a haddock derivation for the whole project combined.... but we can do that later.

Reply:
... maybe we should do it now. Just thinking about what's going to happen if I update to use this:
- Our CI builds all the library etc. components, and
- Our `shellFor` will pull in the doc, so
- The first time people run `nix-shell` or whatever they'll have to build all the docs locally

So we'll really want a way to build all that in CI easily.

Reply:
I think if users want the shell built and cached by CI, they should build the shell in the CI jobset. I do this for cardano-wallet for example (it also needs something to ensure that the build dependencies become runtime dependencies - e.g. `echo ${lib.interperse " " buildInputs} > $out`).

Reply:
Hmm, at the moment I just include all the stuff that's *in* the shell, rather than building the shell itself. I guess I could do that (although I think it's actually nice to be able to explicitly say "and make sure all the docs build").

Reply:
I was able to reproduce the old behaviour which is basically that when `doHaddock = false` the library derivation has no `doc` and `haddockDir` is `null`.  This turns out to be important as we still want a way to turn of haddock for packages on which it does not work and still get haddocks for the rest of the packages.